pkg/cluster/manager/tls.go

```diff
@@ -0,0 +1,182 @@
+	base := metadata.GetBaseMeta()
+
+	// set tls_enabled
+	globalOptions := topo.BaseTopo().GlobalOptions
+	// if force is true, skip this check
+	if globalOptions.TLSEnabled == enable && !gOpt.Force {
+		if enable {
+			m.logger.Infof("cluster `%s` TLS status is already enable\n", name)
+		} else {
+			m.logger.Infof("cluster `%s` TLS status is already disable\n", name)
+		}
+		return nil
+	}
+	globalOptions.TLSEnabled = enable
+
+	if err := checkTLSEnv(topo, name, base.Version, skipConfirm); err != nil {
+		return err
+	}
+
+	var (
+		sshProxyProps *tui.SSHConnectionProps = &tui.SSHConnectionProps{}
+	)
+	if gOpt.SSHType != executor.SSHTypeNone {
+		var err error
+		if len(gOpt.SSHProxyHost) != 0 {
+			if sshProxyProps, err = tui.ReadIdentityFileOrPassword(gOpt.SSHProxyIdentity, gOpt.SSHProxyUsePassword); err != nil {
+				return err
+			}
+		}
+	}
+
+	// delFileMap: files that need to be cleaned up, if flag -- cleanCertificate are used
+	delFileMap, err := getTLSFileMap(m, name, topo, enable, cleanCertificate, skipConfirm)
+	if err != nil {
+		return err
+	}
+
+	// Build the tls  tasks
+	t, err := buildTLSTask(
+		m, name, metadata, gOpt, reloadCertificate, sshProxyProps, delFileMap)
+	if err != nil {
+		return err
+	}
+
+	ctx := ctxt.New(
+		context.Background(),
+		gOpt.Concurrency,
+		m.logger,
+	)
+	if err := t.Execute(ctx); err != nil {
+		if errorx.Cast(err) != nil {
+			// FIXME: Map possible task errors and give suggestions.
+			return err
+		}
+		return perrs.Trace(err)
+	}
+
+	if err := m.specManager.SaveMeta(name, metadata); err != nil {
+		return err
+	}
+
+	if !enable {
+		// the cleanCertificate parameter will only take effect when enable is false
+		if cleanCertificate {
+			os.RemoveAll(m.specManager.Path(name, spec.TLSCertKeyDir))
+		}
+		m.logger.Infof("\tCleanup localhost tls file success")
+	}
+
+	if enable {
+		m.logger.Infof("Enable cluster `%s` TLS between TiDB components successfully", name)
+	} else {
+		m.logger.Infof("Disable cluster `%s` TLS between TiDB components successfully", name)
+	}
+	return nil
+}
+
+// checkTLSEnv check tiflash vserson and show confirm
+func checkTLSEnv(topo spec.Topology, clusterName, version string, skipConfirm bool) error {
+	// check tiflash version
+	if clusterSpec, ok := topo.(*spec.Specification); ok {
+		if clusterSpec.GlobalOptions.TLSEnabled {
+			if semver.Compare(version, "v4.0.5") < 0 && len(clusterSpec.TiFlashServers) > 0 {
+				return fmt.Errorf("TiFlash %s is not supported in TLS enabled cluster", version)
+			}
+		}
+
+		if len(clusterSpec.PDServers) != 1 {
+			return errorx.EnsureStackTrace(fmt.Errorf("Multiple PD nodes is not supported enable/disable TLS")).
+				WithProperty(tui.SuggestionFromString("Please `scale-in` PD nodes to one and try again."))
+		}
+	}
+
+	if !skipConfirm {
+		return tui.PromptForConfirmOrAbortError(
+			fmt.Sprintf("Enable/Disable TLS will %s the cluster `%s`\nDo you want to continue? [y/N]:",
+				color.HiYellowString("restart"),
```

Comment:
```suggestion
				color.HiYellowString("stop and restart"),
```

It would be better if we make it clear that the cluster will be unavailable during the process.